### PR TITLE
Add a `.gitattributes` recommendation to the AssetLib guidelines

### PR DESCRIPTION
**Pull request by pycbouh**
See https://github.com/godotengine/godot-proposals/issues/2981 and comments to https://github.com/godotengine/godot-docs/pull/3240. This question is raised pretty often by plugin developers, so it's well worth mentioning.

### CODE REVIEW

File: community/asset_library/submitting_to_assetlib.rst [hunk 1]
```diff
@@ -75,6 +75,28 @@ library a better place for all users.
...
+  and prevents them from being downloaded by the asset library users.
+  For a typical plugin **.gitattributes** may look like this:
+
+  .. code-block:: none
+
+    /.gitattributes     export-ignore
+    /.gitignore         export-ignore
+    /LICENSE            export-ignore
+    /README.md          export-ignore
+    /project.godot      export-ignore
```
Comment (Calinou) [inline community/asset_library/submitting_to_assetlib.rst:90]: If we are excluding `project.godot`, I would also exclude the icon files which are generally `icon.png` or `icon.svg` (it's fine to have nonexistent paths in `.gitattributes`).

I would also exclude the license file if it has a `.md` extension:

```suggestion
    /.gitattributes     export-ignore
    /.gitignore         export-ignore
    /LICENSE            export-ignore
    /LICENSE.md         export-ignore
    /README.md          export-ignore
    /project.godot      export-ignore
    /icon.png           export-ignore
    /icon.svg           export-ignore
```

**Comment by akien-mga** [comment]
I don't think it's a good idea to encourage exporting without license. It means that people downloading a zip of the repo will not get the license terms, while the license should explicitly always go together with the content it governs.

The real problem is the bad workflow that we impose where people should put the whole `addons/<name>/` in their repo, and thus have top-level files that might conflict with the installation destination.

I've been saying this is a terrible idea from the start, and I think time has proven it right repeatedly.

I think we should fix that workflow instead of working it around like this. A proper workflow would be:
- Plugin repo contains plugin files as top-level.
- Godot installs the plugin automatically in `addons/<name>/<plugin files>`

The name could be configured in `plugin.cfg`.

**Comment by Calinou** [comment]
> I think we should fix that workflow instead of working it around like this. A proper workflow would be:
>
> - Plugin repo contains plugin files as top-level.
> -Godot installs the plugin automatically in addons/<name>/<plugin files>
>
> The name could be configured in `plugin.cfg`.

The issue is that a single repository may want to install multiple plugins, especially since we don't have dependency tracking yet. Also, non-plugin assets still need to have some kind of configuration file to define the folder they should install in.

File: community/asset_library/submitting_to_assetlib.rst (hunk 1 above)
Comment (pycbouh) [reply community/asset_library/submitting_to_assetlib.rst:90]: Seems redundant if you don't actually have those files, but can be added I guess.

**Comment by pycbouh** [comment]
> I don't think it's a good idea to encourage exporting without license.

I don't think my additions encourage that? The second point I've added specifically covers that, though. We can work on its phrasing if you don't find it satisfactory.

> * If you are submitting a plugin, add a copy of your license and readme
  to the plugin folder itself. This is the folder that users are guaranteed to
  keep with their project, so a copy ensures they always have those files handy
  (and helps them fulfill your licensing terms).

> I've been saying this is a terrible idea from the start, and I think time has proven it right repeatedly.

You get no arguing from me here, but implementing a better plugin deployment system is a lot of work that we don't have a contributor to do. And this is what people can actually do now to help with the clutter. And what we, plugin developers, actually have to do and encourage others to do.

**Review by akien-mga — APPROVED** [review]

**Comment by mhilbrunner** [comment]
Thank you!

**Comment by akien-mga** [comment]
Cherry-picked for 3.3.

**Pull request merged**